lib/api/bucketPut.js

```diff
@@ -30,12 +76,24 @@ function _parseXML(request, log, cb) {
                 .LocationConstraint[0];
             log.trace('location constraint',
                 { locationConstraint });
-            return cb(null, locationConstraint);
+            const locationCheck = checkLocationConstraint(request,
+              locationConstraint, log);
+            if (locationCheck.error) {
+                return cb(locationCheck.error);
+            }
+            return cb(null, locationCheck.locationConstraint);
         });
     }
-    // We need the second parameter to be `undefined` so the waterfall maintains
-    // the position of the `next` argument.
-    return process.nextTick(() => cb(null, undefined));
+    // We need the second parameter to be empty string so the waterfall
```

Comment:
I don't think you need this comment anymore. We are sending either the default or an empty string now.